numba/typed/dictobject.py

```diff
@@ -262,46 +265,57 @@ def codegen(context, builder, sig, args):
 
 @intrinsic
 def _dict_set_method_table(typingctx, dp, keyty, valty):
-    """Wrap numba_dict_set_method_table
-    """
+    """Wrap numba_dict_set_method_table"""
     resty = types.void
     sig = resty(dp, keyty, valty)
 
     def codegen(context, builder, sig, args):
-        vtablety = ir.LiteralStructType([
-            ll_voidptr_type,  # equal
-            ll_voidptr_type,  # key incref
-            ll_voidptr_type,  # key decref
-            ll_voidptr_type,  # val incref
-            ll_voidptr_type,  # val decref
-        ])
+        # correspond to type_based_methods_table in dictobject.h
+        vtablety = ir.LiteralStructType(
+            [
+                ll_voidptr_type,  # key equal
+                ll_voidptr_type,  # key copy
+                ll_voidptr_type,  # key zero
+                ll_voidptr_type,  # value copy
+                ll_voidptr_type,  # value zero
+                ll_voidptr_type,  # key incref
+                ll_voidptr_type,  # key decref
+                ll_voidptr_type,  # value incref
+                ll_voidptr_type,  # value decref
+            ]
+        )
         setmethod_fnty = ir.FunctionType(
-            ir.VoidType(),
-            [ll_dict_type, vtablety.as_pointer()]
+            ir.VoidType(), [ll_dict_type, vtablety.as_pointer()]
         )
         setmethod_fn = ir.Function(
             builder.module,
             setmethod_fnty,
-            name='numba_dict_set_method_table',
+            name="numba_dict_set_method_table",
         )
         dp = args[0]
         vtable = cgutils.alloca_once(builder, vtablety, zfill=True)
 
-        # install key incref/decref
+        # install type_based_methods_table
         key_equal_ptr = cgutils.gep_inbounds(builder, vtable, 0, 0)
-        key_incref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 1)
-        key_decref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 2)
-        val_incref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 3)
-        val_decref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 4)
+        key_copy_ptr = cgutils.gep_inbounds(builder, vtable, 0, 1)
+        key_zero_ptr = cgutils.gep_inbounds(builder, vtable, 0, 2)
+        value_copy_ptr = cgutils.gep_inbounds(builder, vtable, 0, 3)
+        value_zero_ptr = cgutils.gep_inbounds(builder, vtable, 0, 4)
+        # install inc/dec refs
+        key_incref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 5)
+        key_decref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 6)
+        value_incref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 7)
+        value_decref_ptr = cgutils.gep_inbounds(builder, vtable, 0, 8)
 
         dm_key = context.data_model_manager[keyty.instance_type]
         if dm_key.contains_nrt_meminfo():
-            equal = _get_equal(context, builder.module, dm_key, 'dict_key')
+            key_equal = _get_container_equal(
+                context, builder.module, dm_key, "dict_key")
```

Comment:
Is this the only place that calls `_get_container_equal`? If so, the `"dict_key"` suffix can be moved to inside the function, as it won't change.

Reply:
yes, it's the only one usage. hide this argument into the function.